rfc/rfc-66/rfc-66.md

```diff
@@ -0,0 +1,165 @@
+the plan does not include file t2_t5_v1.w2, in the reader view, the log file should be assigned to a different file slice than the t4 instant time.
+
+#### Global Monotonically Increasing Timestamp
+
+In order to make the time deterministic among cloud storages, we use a logical time generated by a special **TimeGenerator**, see [HUDI-1623](https://issues.apache.org/jira/browse/HUDI-1623) for details.
+
+#### Assumption On Locks
+In order to get global monotonically increasing time generation, we may introduce lock providers on the instant time generation and the creation of completed metadata file.
+That means, for each instant time/completion time generation request, there is a try-lock action. As a special case, because we store the completion time as part of the completed
+metadata file name, the time generation and file creation should have atomicity altogether, they should be guarded under the same lock of regular time generation request.
+
+### The Log File Naming Convention
+
+We use the current instant time instead of the base commit time in the file name,
+so that tasks from different writers can not conflict in file name. We can also parse the file name to fetch the instant time quickly. 
+Finally, the log name is with the following pattern:
+
+```shell
+${uuid}_${instant}.log.${version}_${task_token}
+```
+
+### The Sorting Rules for Log Files from Different Writers
+
+The sorting rules is important because it somehow decides the natual order processing,
+especially when the event time field are the same, and we don't know which payload to choose when combining.
+Here we can keep using the log file name compactor of current codebase, that is:
+
+```sql
+order by base_commit_time, version_number, write_token
+```
+
+For a single writer, its log files natual order is preserved by the auto increasing version_number,
+for multiple writers, we try the best to preserve the natual order still with the version number,
+but the write_token has deterministic priority in sorting, which breaks the file generation order.
+
+![log file sequence](log_file_sequence.png)
+
+### Format changes
+
+| Type                       | Changes                                                                                                                            |
+|----------------------------|------------------------------------------------------------------------------------------------------------------------------------|
+| Commit/compaction metadata | No changes                                                                                                                         |
+| Commit file name           | we are only going to add completion time in completed metadata filename for all actions                                            |
+| Log file format            | Adding deltacommit instant time to log file name. The file name does not contain base instant time; No changes to log block format |
```

Comment:
i would say let's keep the base instant time too unless there is some inefficiency. 
Otherwise we will have to add the base instatnt time to log file header (or repeated in log block header)

Reply:
> let's keep the base instant time too unless there is some inefficiency.

There is inefficiency because the base instant time needs additional look up to the fs view, which is burdensome for the driver especially for large table with updates to multiple data partitions. This is good chance to remove that, if position deletes needs base instant time, do the very specific fs view look-up and put it in the log blocks.

Reply:
If we include base instant time in the log files, it will become confusing because the logical file-slice will span across base instant-times